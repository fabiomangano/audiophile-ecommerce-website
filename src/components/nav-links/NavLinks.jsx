import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import style from './NavLinks.module.scss';

export const links = [
  {
    to: '/',
    label: 'Home',
  },
  {
    to: '/category/headpones',
    label: 'Headpones',
  },
  {
    to: '/category/speakers',
    label: 'Speakers',
  },
  {
    to: '/category/earpones',
    label: 'Earpones',
  },
];

export const NavLinks = ({ direction }) => {
  let className;

  if (direction === 'horizontal') className = style.links_horizontal;
  if (direction === 'vertical') className = style.links_vertical;

  return (
    <div className={style.container}>
      <ul className={className}>
        {links.map(({ to, label }) => (
          <li key={label}>
            <Link to={to} className={style.link}>
              {label}
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
};

NavLinks.defaultProps = {
  direction: 'horizontal',
};

NavLinks.propTypes = {
  direction: PropTypes.oneOf(['horizontal', 'vertical']),
};

export default NavLinks;
