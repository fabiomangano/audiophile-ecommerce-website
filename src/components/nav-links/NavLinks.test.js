import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import NavLinks, { links } from './NavLinks';

describe('Navlinks', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<NavLinks />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('should render all page links', () => {
    const wrapper = mount(
      <MemoryRouter>
        <NavLinks />
      </MemoryRouter>,
    );

    links.forEach(({ to, label }, position) => {
      expect(wrapper.find('a').at(position).children().text()).toEqual(label);
      expect(wrapper.find('a').at(position).props()).toHaveProperty('href', to);
    });
  });
});
