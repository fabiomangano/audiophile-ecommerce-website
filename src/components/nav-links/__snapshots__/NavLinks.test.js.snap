// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navlinks should render correctly 1`] = `
<div
  className="container"
>
  <ul
    className="links_horizontal"
  >
    <li>
      <Link
        className="link"
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        className="link"
        to="/category/headpones"
      >
        Headpones
      </Link>
    </li>
    <li>
      <Link
        className="link"
        to="/category/speakers"
      >
        Speakers
      </Link>
    </li>
    <li>
      <Link
        className="link"
        to="/category/earpones"
      >
        Earpones
      </Link>
    </li>
  </ul>
</div>
`;
